### PR TITLE
Single dictionary support for OD.

### LINKED ISSUE #87: Consolidate external source drops
We have source drops of external projects in several places:

```
src/external
src/nnvm
subtree/xgboost
deps/src
```

We should consolidate these into a single external directory, and make it clear that these are external source drops and generally shouldn't be modified except to take new source drops of these projects.

### PR DESCRIPTION
**Pull request by srikris**
Fixes #87. Object detectors can now take in an annotation column that's just a single dictionary (if its all just 1 objects)

### CODE REVIEW

File: src/unity/python/turicreate/toolkits/object_detector/_sframe_loader.py [hunk 1]
```diff
@@ -166,6 +170,9 @@ def is_valid(ann):
                     return ok_optional
 
                 # Unchanged boxes, for evaluation
+                if not hasattr(label, '__iter__'):
```
Comment (TobyRoseman) [inline src/unity/python/turicreate/toolkits/object_detector/_sframe_loader.py]: Probably want to remove this before you check it in.

File: src/unity/python/turicreate/toolkits/object_detector/_sframe_loader.py [hunk 2]
```diff
@@ -149,6 +149,10 @@ def _next(self):
             oshape = orig_image.shape
             if self.load_labels:
                 label = row[self.annotations_column]
+                if label == None:
+                    label = []
+                if type(label) == dict:
```
Comment (gustavla) [inline src/unity/python/turicreate/toolkits/object_detector/_sframe_loader.py]: How about an `elif` here? Whenever I see if statements like this I assume both can occur and waste a few cycles realizing they can't.

File: src/unity/python/turicreate/toolkits/object_detector/util/_visualization.py [hunk 3]
```diff
@@ -120,6 +120,10 @@ def draw_bounding_boxes(images, annotations, confidence_threshold=0):
     def draw_single_image(row):
         image = row['image']
         anns = row['annotations']
+        if anns == None:
+            anns = []
+        if type(anns) == dict:
+            anns = [anns]
```
Comment (gustavla) [inline src/unity/python/turicreate/toolkits/object_detector/util/_visualization.py]: Same here, `elif`

File: src/unity/python/turicreate/test/test_object_detector.py [hunk 4]
```diff
@@ -171,6 +171,21 @@ def test_create_with_invalid_annotations_not_dict(self):
...
+                          annotations=self.annotations,
+                          max_iterations=1,
+                          model=self.pre_trained_model)
+
+        pred = dict_model.predict(sf_copy)
+        metrics = dict_model.evaluate(sf_copy)
+        annotated_img = tc.object_detector.util.draw_bounding_boxes(sf_copy[self.feature],
+                sf_copy[self.annotations])
+
+
```
Comment (gustavla) [inline src/unity/python/turicreate/test/test_object_detector.py]: Remove one of these empty lines. /The PEP8 Police

File: src/unity/python/turicreate/toolkits/object_detector/_sframe_loader.py (hunk 2 above)
Comment (srikris) [reply src/unity/python/turicreate/toolkits/object_detector/_sframe_loader.py]: Done

File: src/unity/python/turicreate/toolkits/object_detector/_sframe_loader.py (hunk 1 above)
Comment (srikris) [reply src/unity/python/turicreate/toolkits/object_detector/_sframe_loader.py]: Done.

File: src/unity/python/turicreate/toolkits/object_detector/util/_visualization.py (hunk 3 above)
Comment (srikris) [reply src/unity/python/turicreate/toolkits/object_detector/util/_visualization.py]: Done

**Pull request merged**